### PR TITLE
Make multi-query json tests single-query

### PR DESCRIPTION
**Pull request by thijs-obj**
Improvements for the tests of SeriesJson:
* made multi-query tests single-query
* some other small tweaks

This is preparation for supporting SeriesJson on Athena

### CODE REVIEW

File: bach/tests/functional/bach/test_series_json.py [hunk 1]
```diff
@@ -177,56 +178,56 @@ def test_json_getitem_special_chars(engine, dtype):
...
-                {'id': 'top-10', '_type': 'SectionContext'}, {'id': '5o7Wv5Q5ZE', '_type': 'ItemContext'}]],
-            [4, []]
-        ]
+        expected_columns=['_index_row', 'one_to_end', 'one_to_minus_one', 'end_to_end'],
+        expected_data=expected_data
     )
 
-    bts = bt.mixed_column.json[1:-1]
+
+def test_json_getitem_slice_non_happy_mixed_data(engine, dtype):
```
Comment (thijs-obj) [inline bach/tests/functional/bach/test_series_json.py:226]: couldn't combine this part into the single query of `test_json_getitem_slice`, so split this off in a separate function. This way we at least split the two queries over two separate functions.

File: bach/tests/functional/bach/test_series_json.py [hunk 2]
```diff
@@ -248,47 +249,41 @@ def test_json_getitem_slice(engine, dtype):
 # tests below are for functions kind of specific to the objectiv (location) stack
 def test_json_getitem_query(engine, dtype):
```
Comment (thijs-obj) [inline bach/tests/functional/bach/test_series_json.py:250]: also made this one single query

**Review by jansentom — APPROVED** [review]
looks good to me!

**Review by KathiaBarahona — APPROVED** [review]
lgtm :rocket: nice to optimize this tests

**Pull request merged**